activation/activation.go

```diff
@@ -301,6 +317,39 @@ func (b *Builder) generateProof() error {
 	return nil
 }
 
+func (b *Builder) waitForFirstATX(ctx context.Context) {
+	if b.layerClock.GetCurrentLayer().GetEpoch() == 0 { // genesis miner
+		return
+	}
+	if _, err := b.cdb.GetPrevAtx(b.nodeID); err == nil { // not a new miner
+		return
+	}
+
+	// first atx for this node
+	currentLayer := b.layerClock.GetCurrentLayer()
+	waitTill := (currentLayer.GetEpoch() + 1).FirstLayer()
+	b.log.WithContext(ctx).With().Info("building very first atx",
+		log.Stringer("current_epoch", currentLayer.GetEpoch()),
+		log.Stringer("wait_till", waitTill),
+		log.Stringer("wait_till_epoch", waitTill.GetEpoch()))
+	select {
+	case <-ctx.Done():
+		return
+	case <-b.layerClock.AwaitLayer(waitTill):
+		// this estimate work if the majority of the nodes use poet servers that are configured the same way.
+		// TODO: do better when nodes use different poet services
+		timer := time.NewTimer(time.Second * 35)
```

Comment:
i looked up logs and see why you set 35, but it doesn't make any sense. something is wrong with time/layer computation in go-spacemesh

atx's that target next epoch are published at epoch_start + phase_shift - cycle_gap. so after first layer of the epoch started you need to wait phase_shift - cycle_gap + expected broadcast duration (for fastnet we can set it to 5s)

Reply:
yikes. for some reason i didn't update this to use the values set in config. the 35 sec was done for systest testing :(

Reply:
i will file an issue for the clock for now and set the config values in systest.

Reply:
https://github.com/spacemeshos/go-spacemesh/issues/3508

Reply:
you need to wait longer if that clock bug is not fixed in this pr. if you check miner smesher-27 logs `{namespace="test-piwv", pod="smesher-27-0"}  |= "atxBuilder"` - the first atx is still messed up: 

```
{"L":"INFO","T":"2022-08-28T10:07:52.828-0400","N":"9272a.atxBuilder   ","M":"new atx challenge is ready","node_id":"9272a5bf98750e7d5009db903e1b872d80077c1b1f7db2870cc24bd2e527ceaa","module":"atxBuilder","sessionId":"b52ebb53-04c3-411b-a903-2d98b06f3f57","target_epoch":"4","current_epoch":"4","name":"atxBuilder"}
```

and then it fixes atx in the next epoch:

```
{"L":"INFO","T":"2022-08-28T10:09:16.057-0400","N":"9272a.atxBuilder   ","M":"new atx challenge is ready","node_id":"9272a5bf98750e7d5009db903e1b872d80077c1b1f7db2870cc24bd2e527ceaa","module":"atxBuilder","sessionId":"3ee492d6-a97f-4120-9c5d-88ba224181d6","target_epoch":"6","current_epoch":"5","name":"atxBuilder"}
```

to fix this issue either grace period needs to be adjusted by one more layer duration (so total will be 20s for fastnet), or better waitTill should be equal to nextEpoch.FirstLayer().Add(1)